### ISSUE TITLE
always sign outgoing encrypted messages

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-android#1095]
after #1058 - this simply means passing private key into the encryption function for signing

If the outgoing message (or file) is encrypted, it should also be signed.

**Comment by tomholub** [comment]
will be followed by #214 afterwards

**Comment by DenBond7** [comment]
is #655 a duplicate of this one?

**Comment by tomholub** [comment]
It's supersedes that issue. Originally, we were going to make it configurable - user can choose to sign a message or not. But this issue is simpler. All outgoing encrypted messages _must_ also be signed, without any way to configure it, no change in UI.

**Comment by DenBond7** [comment]
@tomholub Please let me clarify a few things

1. A user has a single private key. it seems the simplest case. Then we have the following

![image](https://user-images.githubusercontent.com/2863246/113878950-bb7e5380-97c2-11eb-842e-568fd198ab86.png)

On Android we have `gpg: Note: sender requested "for-your-eyes-only"`. Should I leave it as is or need to find how to drop it?

2. A user has a few private keys. Currently, If we will find a few pubkey we will return the first (we do a search by email).

https://github.com/FlowCrypt/flowcrypt-android/blob/3b56fc45d525cf3439f65e8ec854d2c77707b711/FlowCrypt/src/main/java/com/flowcrypt/email/security/SecurityUtils.kt#L135

What about signing? Should I use the relevant private key(`pair(prv, pub)`) for our purposes or I can sign with all available prv keys?

**Comment by tomholub** [comment]
> What about signing? Should I use the relevant private key(pair(prv, pub)) for our purposes or I can sign with all available prv keys?

You can find private key for signing with the same method that you use to find public key to attach to emails.

That means, under under the same set of conditions (like account or `from` address), the same key pair is used for signing and for attaching public key - in one case a private key, in one case public key.

Both are very good questions - thanks for asking.

**Comment by DenBond7** [comment]
> I don't think email clients actually implement this feature to make any meaningful difference to the user, I would ignore it (for now).
> 
> But it would be good to file another issue about this, on pgpainless. To see how email clients treat this information, and see if it makes any difference.

Here @vanitasvitae has added changes to prevent producing `for-your-eyes-only` by default. Those changes will be applied when we will apply a new version of `PGPainless`(when it will be released)
